components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/dao/RegistryConfigDAOImpl.java

```diff
@@ -0,0 +1,158 @@
+ */
+
+package org.wso2.carbon.identity.entitlement.dao;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.balana.combine.PolicyCombiningAlgorithm;
+import org.wso2.balana.combine.xacml3.DenyOverridesPolicyAlg;
+import org.wso2.carbon.context.CarbonContext;
+import org.wso2.carbon.identity.entitlement.EntitlementException;
+import org.wso2.carbon.identity.entitlement.EntitlementUtil;
+import org.wso2.carbon.identity.entitlement.PDPConstants;
+import org.wso2.carbon.identity.entitlement.internal.EntitlementServiceComponent;
+import org.wso2.carbon.identity.entitlement.pdp.EntitlementEngine;
+import org.wso2.carbon.registry.core.Collection;
+import org.wso2.carbon.registry.core.Registry;
+import org.wso2.carbon.registry.core.exceptions.RegistryException;
+
+/**
+ * This implementation handles the Global PolicyDAO Combining Algorithm management in the Registry.
+ */
+public class RegistryConfigDAOImpl implements ConfigDAO {
+
+    public static final String POLICY_COMBINING_PREFIX_1 = "urn:oasis:names:tc:xacml:1.0:policy-combining-algorithm:";
+    public static final String POLICY_COMBINING_PREFIX_3 = "urn:oasis:names:tc:xacml:3.0:policy-combining-algorithm:";
+    // The logger that is used for all messages
+    private static final Log log = LogFactory.getLog(RegistryConfigDAOImpl.class);
+    private static final String GLOBAL_POLICY_COMBINING_ALGORITHM = "globalPolicyCombiningAlgorithm";
+    private final String policyDataCollection = PDPConstants.ENTITLEMENT_POLICY_DATA;
+    private final Registry registry;
+
+    public RegistryConfigDAOImpl() {
+
+        int tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
+        registry = EntitlementServiceComponent.getGovernanceRegistry(tenantId);
+    }
+
+    /**
+     * Gets the global policy combining algorithm.
+     *
+     * @return global policy combining algorithm
+     */
+    @Override
+    public PolicyCombiningAlgorithm getGlobalPolicyAlgorithm() {
+
+        String algorithm = null;
+        try {
+            if (registry.resourceExists(policyDataCollection)) {
+                Collection collection = (Collection) registry.get(policyDataCollection);
+                algorithm = collection.getProperty(GLOBAL_POLICY_COMBINING_ALGORITHM);
+            }
+
+            if (algorithm == null || algorithm.trim().isEmpty()) {
+                // Reads the algorithm from entitlement.properties file
+                algorithm = EntitlementServiceComponent.getEntitlementConfig().getEngineProperties().
+                        getProperty(PDPConstants.PDP_GLOBAL_COMBINING_ALGORITHM);
+                log.info("The global policy combining algorithm which is defined in the configuration file, " +
+                        "is used.");
+                try {
+                    return EntitlementUtil.getPolicyCombiningAlgorithm(algorithm);
+                } catch (Exception e) {
+                    log.debug(e);
```

Comment:
why we are logging this, instead of throwing? 
and can we use the warn log if applicable

Reply:
This implementation has been directly ported from: https://github.com/wso2/carbon-identity-framework/blob/53fb513c482a5604c264978e56cfe74292327c27/components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/policy/store/DefaultPolicyDataStore.java#L78
So we cannot throw the exception since that would require changing public methods at the higher level.
Here it is logging the exception that the algorithm specified in the entitlement.properties file is not supported, so we can change it to warn log instead.